### DISCUSSION TITLE
Apply mogrify to all images *except* 1000x1000

### DISCUSSION THREAD

**Discussion by Montreal666** [ImageMagick/ImageMagick#2856] [Help]
I use the following command line on a 12000 jpg files folder (to format mp3 cover art files to 1000x1000):

magick mogrify -fuzz 1% -trim +repage -resize 1000^ -gravity Center -crop 1000x1000+0+0 *.jpg

it works perfectly and does exactly what I want,  but takes about 2 hours to run
only a few files are added each day to the same folder and the command is run daily.
To save some CPU cycles and shorten processing time, I'd like to ignore all files already sized 1000x1000 or only apply to files that were not previously "mogrified"

please note that I can't change the file format as it used by another app and must remain intact.

I'm still new to ImageMagick and could use some assistance/advice on this one;

Thanks a lot for your support,

**Comment by fmw42** [comment]
ImageMagick does not have such a feature in mogrify as far as I know. I suggest you write a script loop and test the size and use a conditional to decide whether to process or not.

What is your platform?

**Comment by Montreal666** [comment]
Windows 10; ImageMagick-7.0.10-Q16-HDRI
would defintitely appreciate some guidance on the script loop,
Thanks for the prompt reply

**Reply by fmw42** [reply]
I would help, but I only script in bash Unix. One of the other (Windows) users will likely give you some help. If you use Windows 10 Unix, then I can provide an example.

**Comment by snibgo** [comment]
> 12000 jpg files folder ... only a few files are added each day

I suggest you change your workflow so new files are processed _before_ they are added to that folder, so you don't need to query 12000 files every day.

Anyhow, you can test each file if you really must. Something like this Windows BAT script:
```
setlocal enabledelayedexpansion

for %%F in (*.jpg) do (

  for /F "usebackq" %%L in (`magick -ping "%%F" ^
    -format "DoIt=%%[fx:%w==1000&&%h==1000?0:1]" ^
    info:`) do set %%L

  if !DoIt!==1 (
    magick "%%F" -fuzz 1%% -trim +repage -resize 1000^^ -gravity Center -crop 1000x1000+0+0 +repage "%%F"
  )

)
```
Obviously, check this carefully on a test directory first before letting it loose on your 12000 files.

**Comment by Montreal666** [comment]
Thank you @snibgo , seems to work so far;

> I suggest you change your workflow so new files are processed before they are added to that folder, so you don't need to query 12000 files every day.

unfortunately the files are created  automatically in the same folder by a secondary app I cannot change that.
I have to either rely on the fact that some files are already 1000x1000  or via the modified date;
would the latter be a better/more efficient option and is it achievable via the batch script?
If feasible, I'll test both and report back results;

Thanks again for your assistance, much appreciated.

**Comment by snibgo** [comment]
Selecting the files by their date, using filesystem facilities, should be much faster than using IM to open every file.

**Comment by Montreal666** [comment]
Thanks for your reply, 
since all files have to stay within the same folder + done unattended, is there an efficient way to do it via the script?
And if so, would you mind providing guidance on the script modifications? Still in the learning curve with scripting.
Really appreciate your assistance,
Thank you,

**Comment by snibgo** [comment]
You can get a list of, for example, all jpg files in the current directory that have changed on or after November 1st 2020:
```
f:\web\im>xcopy *.jpg qwezxc\ /D:11-1-2020 /L
F:4096.jpg
F:starryNight.jpg
F:toes_PackEventCombo-9x6.jpg
F:x_PackEventCombo-9x6.jpg
4 File(s)
```
See "help xcopy" for its options.

A script could read this list, ignoring the final line "4 File(s)", and process just those files.
